From: Dimitri Sivanich <sivanich@sgi.com>
Subject: x86: Also move detect_extended_topology before init_intel_cacheinfo(c) in init_intel()
References: bnc#464329
Patch-Mainline: yes or at least submitted

This patch belongs to patches.arch/x86_fix_llc_shared_map__cpu_llc_id_anomolies.patch.
This one does the same change for 64 bit. Above patch is doing things for intel.c,
intel_64.c has been forgotten.

Signed-off-by: Thomas Renninger <trenn@suse.de>

---
 arch/x86/kernel/cpu/intel_64.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/arch/x86/kernel/cpu/intel_64.c
+++ b/arch/x86/kernel/cpu/intel_64.c
@@ -67,6 +67,7 @@ static void __cpuinit srat_detect_node(v
 
 static void __cpuinit init_intel(struct cpuinfo_x86 *c)
 {
+	detect_extended_topology(c);
 	init_intel_cacheinfo(c);
 	if (c->cpuid_level > 9) {
 		unsigned eax = cpuid_eax(10);
@@ -94,7 +95,6 @@ static void __cpuinit init_intel(struct 
 		set_cpu_cap(c, X86_FEATURE_REP_GOOD);
 	set_cpu_cap(c, X86_FEATURE_LFENCE_RDTSC);
 
-	detect_extended_topology(c);
 	if (!cpu_has(c, X86_FEATURE_XTOPOLOGY))
 		c->x86_max_cores = intel_num_cpu_cores(c);
 
